packages/core/src/Util/Retry.ts

```diff
@@ -36,7 +36,7 @@ export async function retry(
 	try {
 		return await functionToRetry(...args);
 	} catch (err) {
-		logger.debug(`error on ${functionToRetry.name}: ${err} `);
+		logger.debug(`error on ${functionToRetry.name}: ${JSON.stringify(err)} `);
```

Comment:
Thanks for the PR!  Have you been able to validate this?

In my experience, `JSON.stringify(err)` usually returns `{}`:

```js
JSON.stringify(new Error("Message"))
// "{}"
```

Whereas this is usually what I have to do:

```suggestion
		logger.debug(`error on ${functionToRetry.name}: ${err.message} `);
```

Reply:
Thanks for feedback, what is happening here is not an error object, it's returning graphql query object that failed, if you check this file in line 33 it's already handled this way. err.message in this case will return undefined